### ISSUE TITLE
Debug command doesn't work

### ISSUE THREAD

**Issue by thefallentree** [vmware/vic#3573]
```
INFO[2017-01-05T11:17:33-08:00] vSphere password for administrator:
INFO[2017-01-05T11:17:36-08:00] ### Configuring VCH for debug ####
INFO[2017-01-05T11:17:37-08:00]
INFO[2017-01-05T11:17:37-08:00] VCH ID: VirtualMachine:vm-160938
WARN[2017-01-05T11:17:38-08:00] Disabling 4A9535A7-96F2-4166-A485-916E89982B76 vpxd.log collection (ServerFaultCode: A specified parameter was not correct.
key)
INFO[2017-01-05T11:17:38-08:00]
INFO[2017-01-05T11:17:38-08:00] Installer version: v0.8.0-7315-c8ac999
INFO[2017-01-05T11:17:38-08:00] VCH version: v0.8.0-7315-c8ac999
time=2017-01-05T11:17:38.708718845-08:00 level=error msg=op=28243.1: Unable to enable SSH in appliance VM: ServerFaultCode: The guest authentication being used does not have sufficient permissions to perform the operation.
time=2017-01-05T11:17:38.708758354-08:00 level=error msg=op=28243.1: Unable to enable ssh on the VCH appliance VM: Unable to enable SSH in appliance VM: ServerFaultCode: The guest authentication being used does not have sufficient permissions to perform the operation.
ERRO[2017-01-05T11:17:38-08:00] Unable to enable SSH in appliance VM: ServerFaultCode: The guest authentication being used does not have sufficient permissions to perform the operation.
ERRO[2017-01-05T11:17:38-08:00] --------------------
ERRO[2017-01-05T11:17:38-08:00] vic-machine-linux failed: Debug failed
```

**Comment by mdubya66** [comment]
@thefallentree Are you sure the username and password have the correct permissions?

ERRO[2017-01-05T11:17:38-08:00] Unable to enable SSH in appliance VM: ServerFaultCode: The guest authentication being used does not have sufficient permissions to perform the operation.

Seems to indicate they do not.

Please attach the full output and if possible the logs from the vic-admin

**Comment by mdubya66** [comment]
please attach the logs from this VCH See https://vmware.github.io/vic-product/assets/files/html/0.8/vic_admin/log_bundles.html for instructions on gathering them from the VCH Admin portal

**Comment by thefallentree** [comment]
@mdubya66  I am using the administrator account, which has full privilege .

**Comment by thefallentree** [comment]
any idea how to debug further? I can't enable debug mode, so I can't login into the machine to do more inspections.

**Comment by mdubya66** [comment]
@emlin can you take a look at this please?

**Comment by emlin** [comment]
@thefallentree The VM remote access and console login are disabled by default for security consideration. But if the vic services are started, you can login to vic-admin to get more information. https://vmware.github.io/vic-product/assets/files/html/0.8/vic_admin/access_vicadmin.html is the instruction for how to do it.

If unfortunately, the vic-admin is not started successfully, we can only enable SSH through vic-machine debug, to get information from inside of the VM.

In theory, if you have full permission, it should be able to do it.
If you don't mind, you could double check the permission for "Virtual machine"->"Guest operations"->"Guest operation program execution"
If the permission is right, you might try to restart the VCH VM, and then retry the vic-machine debug command.

@hickeng in case you ever saw such kind of issue.

**Comment by thefallentree** [comment]
Like I said I am using the administrator account. With full privileges.
Restart doesn't work either

**Comment by emlin** [comment]
Could you access vic-admin?

**Comment by fdawg4l** [comment]
@thefallentree Can you confirm the URL you're providing `vic-machine` is to your VC instance and not your host?  Also can you confirm the username and password you're passing is correct by logging into your VC instance?  It may not be the issue but the error reported leads me to believe it is.

**Comment by thefallentree** [comment]
Yes, they are all correct

**Comment by kakakikikeke** [comment]
Hi, 

I also have the same problem.

Internally it seems that StartProgramInGuest of govmomi can not be executed.
It seems that you use types.NamePasswordAuthentication as authentication data of StartProgramInGuest, but is this cause ?
I guess that NamePasswordAuthentication can not be used because the password of the guest user is not set on the VCH.

Of course, the vCenter user who is executing has administrator privilege.

**Comment by kakakikikeke** [comment]
Hi,

A great news for me that v0.9.0 is newly released, and I tried the debug command again.
Then the debug command succeeded.

A succeeded debug command is

```
./vic-machine-linux debug --target xxx.xxx.xxx.xxx/dc --user admin@vsphere.local --password xxxxxxxxx --compute-resource cluster1/rp1 --thumbprint 50:24:DB:63:97:35:29:24:1B:4B:E3:2A:79:F6:20:D6:02:C7:80:95 --enable-ssh --name "vch01" --rootpw "rootpw"
```

and console log are

```
Mar  8 2017 17:23:27.377+09:00 INFO  ### Configuring VCH for debug ####
Mar  8 2017 17:23:27.979+09:00 INFO  Validating target
Mar  8 2017 17:23:28.254+09:00 INFO
Mar  8 2017 17:23:28.255+09:00 INFO  VCH ID: VirtualMachine:vm-208
Mar  8 2017 17:23:28.266+09:00 INFO
Mar  8 2017 17:23:28.266+09:00 INFO  Installer version: v0.9.0-0-f5d2542
Mar  8 2017 17:23:28.266+09:00 INFO  VCH version: v0.9.0-0-f5d2542
Mar  8 2017 17:23:28.427+09:00 INFO
Mar  8 2017 17:23:28.427+09:00 INFO  SSH to appliance:
Mar  8 2017 17:23:28.427+09:00 INFO  ssh root@xxx.xxx.xxx.xxx
Mar  8 2017 17:23:28.427+09:00 INFO
Mar  8 2017 17:23:28.427+09:00 INFO  VCH Admin Portal:
Mar  8 2017 17:23:28.427+09:00 INFO  https://xxx.xxx.xxx.xxx:2378
Mar  8 2017 17:23:28.427+09:00 INFO
Mar  8 2017 17:23:28.427+09:00 INFO  Published ports can be reached at:
Mar  8 2017 17:23:28.427+09:00 INFO  xxx.xxx.xxx.xxx
Mar  8 2017 17:23:28.427+09:00 INFO
Mar  8 2017 17:23:28.427+09:00 INFO  Docker environment variables:
Mar  8 2017 17:23:28.427+09:00 INFO  DOCKER_HOST=xxx.xxx.xxx.xxx:2375
Mar  8 2017 17:23:28.427+09:00 INFO
Mar  8 2017 17:23:28.427+09:00 INFO  Connect to docker:
Mar  8 2017 17:23:28.427+09:00 INFO  docker -H xxx.xxx.xxx.xxx:2375 info
Mar  8 2017 17:23:28.427+09:00 INFO  Completed successfully
```

I tried it with a combination of v0.9.0 and vcenter 6.5 successfully.
It failed with v0.8.0 and vsphere 6.0u2, it may be due to a combination of versions.

**Comment by mdubya66** [comment]
Not sure if this is related but see the comment in #4214 (https://github.com/vmware/vic/issues/4214#issuecomment-292301925)

**Comment by hickeng** [comment]
We are not performing credential checking in-guest on the VCH appliance - it's only contingent on the user having the correct credentials in vSphere to issue the Guest operation in the first place.

I can only assume that somehow we are defaulting to this message - the code looks that might occur if we cannot unmarshal the body of the request for some reason. If the `authorized_keys` file supplied is too long (I'm not sure what the limit would be) then perhaps that would occur [speculation]. I have happily used authorized_keys from systems with several large keys in them, but not one with the 10s - 100s that sometimes occur.

If this is still happening for anyone, please could you supply a log bundle - we're interested in the init.log primarily.

**Comment by mdubya66** [comment]
Resolved in #4214
